app/assets/stylesheets/components/toggles.scss

```diff
@@ -0,0 +1,69 @@
+.c-toggle {
+  --gap: 2px;
+  width: 2.75em;
+  height: 1.5em;
+  display: inline-flex;
+  cursor: pointer;
+  position: relative;
+
+  $this: '.c-toggle';
+
+  &__control {
+    width: 100%;
+    height: 100%;
+    display: inline-flex;
+    background: var(--toggle-rail-bg);
+    border-radius: 1000px;
+    transition: var(--transition-props);
+
+    @media (prefers-reduced-motion: reduce) {
+      transition-duration: 0ms;
+    }
+
+    &::after {
+      content: '';
+      background: var(--toggle-knob-bg);
+      position: absolute;
+      inset: var(--gap);
+      aspect-ratio: 1;
+      height: calc(100% - (var(--gap) * 2));
+      border-radius: 1000px;
+      transition: var(--transition-props);
+      box-shadow: 0 0 0 1px rgba(var(--black), 0.05),
+        0 2px 3px rgba(var(--black), 0.2);
+
+      @media (prefers-reduced-motion: reduce) {
+        transition-duration: 0ms;
+      }
+    }
+  }
+
+  [type='checkbox'] {
+    position: absolute;
+    inset: 0;
+    width: 100%;
+    height: 100%;
+    cursor: pointer;
+    z-index: var(--z-elevate);
+    opacity: var(--opacity-0);
+    margin: 0;
+
+    &:checked + #{$this}__control {
+      background: var(--toggle-rail-checked-bg);
+
+      &::after {
+        transform: translateX(100%);
+      }
+    }
+
+    &:focus-visible + #{$this}__control {
```

Comment:
```suggestion
    &:focus + #{$this}__control {
```

I'm pretty sure the implementation in latest Chrome/Firefox  for `focus` is the same as `focus-visible` now, whereas Safari still hasn't fully rolled our `focus-visible` support (nearly!)

Reply:
This has been shipped in the latest stable version of Safari not that long time ago so it's now officially available everywhere. So now the question is: how much do we care about earlier versions?

Reply:
I had thought that too, but then [caniuse said it's still only TP](https://caniuse.com/?search=focus-visible) 🤔

Reply:
I guess caniuse is just a little out of date. I think we're all good then, we don't really tend to support older browser versions, and probably apple ecosystem is more likely to be on latest version than other browser users 😄

Reply:
<img width="1401" alt="Zrzut ekranu 2022-04-7 o 14 23 58" src="https://user-images.githubusercontent.com/108287/162197844-fce0de4f-f4fc-45a6-840a-90f24e1a2802.png">

🧐

Reply:
Ha! I don't know what I was looking at then. Maybe my brain just skipped to latest and ignored 15.4? Who can say 😂  Nothing to see here, move along...